src/htm/algorithms/SDRClassifier.cpp

```diff
@@ -58,7 +58,11 @@ PDF Classifier::infer(const SDR & pattern) const {
   PDF probabilities( numCategories_, 0.0f );
   for( const auto bit : pattern.getSparse() ) {
     for( size_t i = 0; i < numCategories_; i++ ) {
-      probabilities[i] += weights_[bit][i];
+      if (weights_.size() > bit) {
+        if (weights_[bit].size() > i) {
```

Comment:
the sign is incorrect, should be `<`. Just leave it without the check (for speed) and optionally add an NTA_ASSERT for the correct dimensions above the loop.

Reply:
I found the NTA_ASSERT dimensions check.  but is not works。

I change to NTA_CHECK